packages/@lightningjs/ui-components/src/components/Keyboard/Keyboard.d.ts

```diff
@@ -32,12 +32,91 @@ export type KeyboardStyle = {
   inputStyle: inputStyleObject;
 };
 
+export type KeyboardFormat = Record<
+  string,
+  Array<Array<string | Record<string, unknown>>>
+>;
+
+declare namespace Keyboard {
+  export interface TemplateSpec extends Base.TemplateSpec {
+    // properties
+    formats?: KeyboardFormat[];
+    centerKeyboard?: boolean;
+    rowWrap?: boolean;
+    centerKeys?: boolean;
+    keyComponent?: lng.Component;
+
+    // custom setters
+    get defaultFormat(): string;
+    get columnCount(): number;
+    get rowCount(): number;
+  }
+}
+
+export type KeyboardSelectedKeyPosition = {
+  row: number;
+  column: number;
+};
+
 export default class Keyboard extends Base {
+  // Properties
+  /**
+   * object containing arrays that represent different formats that the keyboard can be presented in. These arrays can contain strings or objects.
+   */
+  formats?: KeyboardFormat[];
+
+  /**
+   * center the keyboard within it's set width (must set the w property of Keyboard)
+   */
   centerKeyboard?: boolean;
+
+  /**
+   * wrap keyboard rows when navigating left and right
+   */
+  rowWrap?: boolean;
+
+  /**
+   * center the keys within it's set width (must set the w property of Keyboard)
+   */
   centerKeys?: boolean;
-  formats?: lng.Component | Array<lng.Component>;
+
+  /**
+   * component to display for each key
+   */
   keyComponent?: lng.Component;
-  rowWrap?: boolean;
+
   get style(): KeyboardStyle;
   set style(v: StylePartial<KeyboardStyle>);
+
+  // Custom setters
+  get defaultFormat(): string;
+  get columnCount(): number;
+  get rowCount(): number;
+
+  // methods
+  /**
+   * switches the keyboard format used to a newly specified format
+   * @param KeyboardFormat keyboard format to switch to
+   */
+  $toggleKeyboard(next: KeyboardFormat): void;
+
+  /**
+   *
+   * @param keyboard an element rendering a keyboard
+   * @param KeyboardSelectedKeyPosition an object specifying the row and column of the key to select
+   */
+  selectKeyOn(
+    keyboard: lng.Component,
+    KeyboardSelectedKeyPosition: KeyboardSelectedKeyPosition
+  ): void;
+
+  /**
+   * returns an object specifying the row and column of the selected key
+   */
+  getSelectedKey(): KeyboardSelectedKeyPosition;
+
+  /**
+   * An event that, when triggered, calls a method that forces the component to update.
+   */
+  $itemChanged(): void;
```

Comment:
I'm not sure we should refer to these as events since that's a different thing. I'm working on a way to properly type fireAncestors calls but for now perhaps we just note that this is triggered by fireAncestors calls and leave it at that